crates/next-api/src/operation.rs

```diff
@@ -85,20 +91,16 @@ fn wrap_route(route: &Route, entrypoints: OperationVc<Entrypoints>) -> RouteOper
     }
 }
 
-#[turbo_tasks::function]
-fn wrap_endpoint(
-    endpoint: Vc<Box<dyn Endpoint>>,
+/// Given a resolved `Endpoint` and the `Entrypoints` operation that it comes from, connect the
+/// operation and return a `OperationVc` of the `Entrypoint`. This `Endpoint` operation will keep
+/// the entire `Entrypoints` operation alive.
+#[turbo_tasks::function(operation)]
+fn wrap(
+    endpoint: ResolvedVc<Box<dyn Endpoint>>,
     op: OperationVc<Entrypoints>,
 ) -> Vc<Box<dyn Endpoint>> {
     let _ = op.connect();
-    endpoint
-}
-
-fn wrap(
-    endpoint: Vc<Box<dyn Endpoint>>,
-    op: OperationVc<Entrypoints>,
-) -> OperationVc<Box<dyn Endpoint>> {
-    OperationVc::new(wrap_endpoint(endpoint, op))
+    *endpoint
 }
```

Comment:
On my original PR this caused some next.js test failures as it reports Issues that occur in get_entrypoints on every Endpoint (as it wraps over the operation).

Maybe we need to "hide" the issues by calling `take_collectibles()` on the Entrypoints operation. Best in a separate turbo-tasks function to do it only once (it's a bit expensive).